admin/components/Layout/Navigation.tsx

```diff
@@ -5,15 +5,26 @@ import {
   SupportIcon,
   UsersIcon,
 } from '@heroicons/react/outline'
+import useTranslation from 'next-translate/useTranslation'
+import { SVGProps } from 'react'
+interface navigation {
+  title: string
+  href: string
+  icon: (props: SVGProps<SVGSVGElement>) => JSX.Element
+}
+
+export const navigations = () => {
+  const { t } = useTranslation()
```

Comment:
Dobry pomysł, jedna uwaga:
W Reactcie jest taka konwencja, żeby hooki wywoływać tylko w komponentach lub w customowych hookach. Customowe hooki to z grubsza zwykłe funkcje wywołujące inne hooki ;) 
Więcej: 

- https://reactjs.org/docs/hooks-rules.html
- https://reactjs.org/docs/hooks-custom.html

Zainstalowałem plugin do Eslinta, który sprawdza reguły zgodnie z dobrymi praktykami. Trzeba zmienić nazwę tej funkcji, żeby zaczynała się od "use" -> `useNavigation` albo `useRoutes` to moje propozycje

BTW ogólnie powiedziałbym, że dobrą praktyką jest używanie jakiegoś czasownika w nazwach funkcji.